## <center>Практическое задание к лекции 3.

**1. Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean):**  
   **а) среднее арифметическое,**  
   **б) среднее квадратичное отклонение,**  
   **с) смещенную и несмещенную оценки дисперсий для данной выборки.**

In [368]:
salaries = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
salaries

array([100,  80,  75,  77,  89,  33,  45,  25,  65,  17,  30,  24,  57,
        55,  70,  75,  65,  84,  90, 150])

Cреднее арифметическое

In [369]:
salaries_mean = salaries.sum() / salaries.size

In [370]:
print(f'Среднее арифметическое для данной выборки М(Х) = {salaries_mean: .2f}')

Среднее арифметическое для данной выборки М(Х) =  65.30


Cреднее квадратичное отклонение

In [371]:
S=(np.sum((salaries - salaries_mean)**2) / salaries.size)**0.5

In [372]:
print(f'Среднее квадратичное отклонение = {S: .4f}')

Среднее квадратичное отклонение =  30.8239


Cмещенная и несмещенная дисперсии

In [373]:
q=np.sum((salaries - salaries_mean)**2) / salaries.size
q1=np.sum((salaries - salaries_mean)**2) / (salaries.size - 1)

In [374]:
print(f'Cмещенная дисперсия = {q: .4f}')
print(f'Несмещенная дисперсия = {q1: .4f}')

Cмещенная дисперсия =  950.1100
Несмещенная дисперсия =  1000.1158


---  
#### Ответ:  
а) $М(Х)= 65.30$;  
б) $\sigma = 30.8239$;  
в) $\sigma^2_{\text{смещ}}=950.1100$, $\sigma^2_{\text{несмещ}}=1000.1158$.  

---

**2. В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?**

In [375]:
from math import factorial

def combinations(n,k):
    return factorial(n)/(factorial(k)*factorial(n-k))

Определим вероятность события $A_1$:

In [376]:


P1=(combinations(3,2)/combinations(8,2))*(combinations(5,3)*combinations(7,1)/combinations(12,4))


In [377]:
print(f'Вероятность события "из первой корзины не вытащили ни одного белого мяча, из второй вытащили 3 белых мяча"\nP(A1) = {P1:.4f}')

Вероятность события "из первой корзины не вытащили ни одного белого мяча, из второй вытащили 3 белых мяча"
P(A1) = 0.0152


Определим вероятность события $A_2$:

In [378]:
P2=(combinations(5,1)*combinations(3,1)*combinations(5,2)*combinations(7,2))/(combinations(8,2)*combinations(12,4))

In [379]:
print(f'Вероятность события "из первой корзины вытащили 1 белый мяч, из второй вытащили 2 белых мяча"\nP(A2) = {P2:.4f}'
     )

Вероятность события "из первой корзины вытащили 1 белый мяч, из второй вытащили 2 белых мяча"
P(A2) = 0.2273


Определим вероятность события $A_3$:

In [380]:
P3=(combinations(5,2)*combinations(5,1)*combinations(7,3))/(combinations(8,2)*combinations(12,4))

In [381]:
print(f'Вероятность события "из первой корзины вытащили 2 белых мяча, из второй вытащили 1 белый мяч"\nP(A2) = {P3:.4f}'
     )

Вероятность события "из первой корзины вытащили 2 белых мяча, из второй вытащили 1 белый мяч"
P(A2) = 0.1263


Вычислим $P(A)$

In [382]:
print(f'Вероятность того, что из вытащенных мячей 3 белые Р(А) = {P1+P2+P3: .4f}')

Вероятность того, что из вытащенных мячей 3 белые Р(А) =  0.3687


---  
#### Ответ:  
Вероятность того, что 3 мяча белые $P(A)\approx 36.87\%$  

---

**3. На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен:**  
   **a) первым спортсменом;**  
   **б) вторым спортсменом;**  
   **в) третьим спортсменом.**

In [383]:
A = 1/3 * 0.9 + 1/3 * 0.8 + 1/3 * 0.6
P1=(1/3 * 0.9) / A
P2=(1/3 * 0.8) / A
P3=(1/3 * 0.6) / A

In [384]:
print(f'Вероятность того, что выстрел произвёл первый спортсмен: {P1:.4f};\n'
      f'Вероятность того, что выстрел произвёл второй спортсмен: {P2:.4f};\n'
      f'Вероятность того, что выстрел произвёл третий спортсмен: {P3:.4f}.')


Вероятность того, что выстрел произвёл первый спортсмен: 0.3913;
Вероятность того, что выстрел произвёл второй спортсмен: 0.3478;
Вероятность того, что выстрел произвёл третий спортсмен: 0.2609.


---
#### Ответ:  
а) Вероятность того, что выстрел произвёл первый спортсмен:  $P_{A}(B_1)\approx39.13\%$;  
б) Вероятность того, что выстрел произвёл второй спортсмен:  $P_{A}(B_2)\approx34.78\%$;  
в) Вероятность того, что выстрел произвёл третий спортсмен:  $P_{A}(B_3)\approx26.09\%$.    

---

**4. В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится:**  
   **a) на факультете A;**  
   **б) на факультете B;**    
   **в) на факультете C?**  

In [385]:
A = 1/4 * 0.8 + 1/4 * 0.7 + 1/2 * 0.9
P1=(1/4 * 0.8) / A
P2=(1/4 * 0.7) / A
P3=(1/2 * 0.9) / A

In [386]:
print(f'Вероятность того, что студент учится на факультете А: {P1:.4f};\n'
      f'Вероятность того, что студент учится на факультете B: {P2:.4f};\n'
      f'Вероятность того, что студент учится на факультете C: {P3:.4f}.'
      )

Вероятность того, что студент учится на факультете А: 0.2424;
Вероятность того, что студент учится на факультете B: 0.2121;
Вероятность того, что студент учится на факультете C: 0.5455.


---  
#### Ответ:  
Вероятность того, что студент учится на факультете А:  $\approx24.24\%$;  
Вероятность того, что студент учится на факультете B:  $\approx21.21\%$;  
Вероятность того, что студент учится на факультете C:  $\approx54.55\%$.  

---

**5. Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя:**  
   **а) все детали;**  
   **б) только две детали;**  
   **в) хотя бы одна деталь;**  
   **г) от одной до двух деталей?**

In [387]:
P1=0.1*0.2*0.25
P2=0.1*0.2*0.75+0.1*0.25*0.8+0.2*0.25*0.9
P3=0.9*0.8*0.75
P3_1=1-P3
P4=0.1*0.8*0.75+0.2*0.9*0.75+0.25*0.9*0.8

In [388]:
print(f'Вероятность того, что из строя выйдут все детали Р(3) = {P1:.4f}')
print(f'Вероятность того, что из строя выйдут только 2 детали Р(2) = {P2:.4f}')
print(f'Вероятность того, что из строя не выйдет ни одной детали Р(0) = {P3:.4f}')
print(f'Вероятность того, что выйдет из строя хотя бы одна деталь Р(>=1) = {P3_1:.4f}')
print(f'Вероятность того, что выйдет из строя выйдут от 1 до 2 деталей Р(1,2) = {P4+P2:.4f}')

Вероятность того, что из строя выйдут все детали Р(3) = 0.0050
Вероятность того, что из строя выйдут только 2 детали Р(2) = 0.0800
Вероятность того, что из строя не выйдет ни одной детали Р(0) = 0.5400
Вероятность того, что выйдет из строя хотя бы одна деталь Р(>=1) = 0.4600
Вероятность того, что выйдет из строя выйдут от 1 до 2 деталей Р(1,2) = 0.4550
